### DISCUSSION TITLE
Ender 3 V3 source code

### DISCUSSION THREAD

**Discussion by NordicPyro** [CrealityOfficial/Creality_Sonic_Pad#89] [General]
Apparently only precompiled binaries are downloadable at the moment. Would be nice if we had access to the source code.

**Comment by schnoog** [comment]
![image](https://github.com/CrealityOfficial/Creality_Sonic_Pad/assets/1537705/070eb3af-9fdd-46cf-91b5-2295e8e0d650)

>this model does not support "open source"<
Seems somebody in China has real problems with understanding the GPL

**Comment by ShixinSun** [comment]
They not only have problems understanding open source and its licenses, but also how software development works in general. 

When I asked for the current source code for the Ender 3 S1, I got this answer.

![cs_1](https://github.com/CrealityOfficial/Creality_Sonic_Pad/assets/158030162/eaf0e7e6-6a2b-4929-9ed6-ce51cd83408c)

**Reply by schnoog** [reply]
Just imagine somebody would explain the usage of github to them. 

Or is there a special plan for which they have to pay per commit????

**Comment by CalDymos** [comment]
They always delete the source code after compiling! ;-)

Quote Creality Report:
> Dear Customer
> 
> 
> Thank you for contacting Creality customer support.
> 
> 
> We don’t have more open source files, only the most basic ones, because everything is modified on the most basic files.
> 
> 
> Thank you!

**Reply by schnoog** [reply]
Hard that the Chinese education system fails to give them the ability to read the GPL. 

With this statement (it's modified but not open source) they say clealy: We don't care about open source, licences and law.

My only conclusion after 4 years of buying creality products: Never again Creality, the lawless company.

**Comment by psycho-neon** [comment]
@CrealityTech when?

**Comment by liberodark** [comment]
Hi,

I think you can share the source of this printer now ?

Best Regards

**Comment by rubienr** [comment]
I see so many posts about Creality violating license agreements, but what can we do to enforce this? 
Does sb. know if platforms like 

- https://sfconservancy.org/
- https://www.fsf.org

have already been contacted? Are those the correct ones to approach in this regard?

Can sb. recommend other platforms/institutions we can approach?
Doe anyone have experience with such cases?

Kind regards!

**Comment by marecl** [comment]
So SE and KE can get some alternatives but base V3 can't? I regret buying this printer even more now
